src/main/java/com/ldtteam/structurize/management/StructureName.java

```diff
@@ -106,13 +109,14 @@ else if (name.startsWith(Structures.SCHEMATICS_CACHE + Structures.SCHEMATICS_SEP
         {
             hut = schematic.split("\\d+")[0];
             section = Structures.SCHEMATICS_PREFIX;
-            if (ForgeRegistries.BLOCKS.getValue(new ResourceLocation(Constants.MINECOLONIES_MOD_ID, "blockhut" + hut)) != Blocks.AIR || HUTS.contains(hut))
-            {
-                section = hut;
-            }
-            else
-            {
-                hut = "";
+            //This allows mods other than minecolonies to create hut buildings and use the buildtool with them.
+            for (final Map.Entry<String, ModFileInfo> origin : StructureLoadingUtils.getOriginMods().entrySet()) {
```

Comment:
Seems okay, just the formatting is totally off.

Reply:
What about the formatting needs fixed? Other than my somewhat unnecessary brackets? (former C developer, I'm a bracket addict.

Reply:
I think I see it now. I'm dumb

Reply:
Wonder where the setting in Intellij that is f-ing with that is...